### PR TITLE
Add actions for building and testing

### PR DESCRIPTION
**Pull request by ljusten**
Adds actions to build and test the code on both Windows and Linux.

### CODE REVIEW

**Review by wurwunchik — CHANGES_REQUESTED** [review]
As shortly mentioned, maybe it would make sense to remove 30 from the targets as well.

File: data_store/data_provider.h [hunk 1]
```diff
@@ -105,6 +105,9 @@ class DataProvider : public DataStoreReader {
   // Periodically cleans up data in |writer_|.
   void CleanupThreadMain() ABSL_LOCKS_EXCLUDED(shutdown_mutex_, cleaned_mutex_);
 
+  // Returns the current time of |steady_clock_| in seconds.
+  int64_t GetNowSec();
```
Comment (chrschng) [inline data_store/data_provider.h:109]: Nit: `GetSteadyNowSec`

File: data_store/data_provider.h [hunk 2]
```diff
@@ -124,15 +127,18 @@ class DataProvider : public DataStoreReader {
...
+  // compilers we can't use std::chrono::time_point with atomics, so keep the
+  // time in seconds.
+  std::atomic<int64_t> last_access_sec_;
 
   // Identifies if new data was added to the cache since the last cleanup.
   std::atomic<bool> chunks_updated_;
 
   // Clock to track the last access time.
   SteadyClock* steady_clock_ = DefaultSteadyClock::GetInstance();
+  std::chrono::time_point<std::chrono::steady_clock> first_now_ =
```
Comment (chrschng) [inline data_store/data_provider.h:140]: Nit: Can this be made `const`?

**Comment by ljusten** [comment]
> As shortly mentioned, maybe it would make sense to remove 30 from the targets as well.

I would suggest to do renamings when we merge asset_stream_manager into cdc_stream. Then we can call everything cdc_stream.

**Review by chrschng — CHANGES_REQUESTED** [review]
This is cool, thanks!

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+      - name: Initialize submodules
+        run: git submodule update --init --recursive
+
+      - name: Build (fastbuild)
+        run: |
+          bazel build --config=windows `
+                      //cdc_rsync //asset_stream_manager //tests_common //tests_asset_streaming_30 //tests_cdc_rsync
+
+      - name: Test (fastbuild)
```
[the remark this answers is not in the document]
Comment (chrschng) [reply .github/workflows/build_and_test.yml:49]: The command above is correct as I wrote it. It includes all test targets `(//...)`, then exclude stuff that we don't want to test (`//third_party`) or stuff we don't want to test twice (`//tests_*`).

**Comment by chrschng** [comment]
<yfd

File: data_store/data_provider.h (hunk 2 above)
Comment (ljusten) [reply data_store/data_provider.h:140]: Done.

File: data_store/data_provider.h (hunk 1 above)
Comment (ljusten) [reply data_store/data_provider.h]: Done.

**Comment by chrschng** [comment]
The Github UI freaks me out. 🙈

File: .github/workflows/build_and_test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ljusten) [reply .github/workflows/build_and_test.yml:49]: Ah got it. Done. Though the tests* are binaries, not tests, so they don't have to be excluded.

**Review by chrschng — DISMISSED** [review]
Thanks!

**Review by chrschng — APPROVED** [review]

**Comment by wurwunchik** [comment]
Let's do renaming later!

**Review by wurwunchik — APPROVED** [review]

**Comment by ljusten** [comment]
Sorry, had to make another change. Turns out disk_data_store is needed for Linux *facepalm*

**Review by chrschng — DISMISSED** [review]

**Review by chrschng — DISMISSED** [review]

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        run: |
+          bazel build --config=linux -- //... -//third_party/...
+
+      # Skip file_finder_test: The test works when file_finder_test is run
+      # directly, but not through bazel test. The reason is, bazel test
+      # creates symlinks of test files, but the finder ignores symlinks.
+      # Also run with --jobs 1 since some tests write to a common tmp dir.
+      - name: Test (fastbuild)
+        run: |
+          bazel test --config=linux --test_output=errors --jobs 1 //... -//third_party/... -//cdc_rsync_server:file_finder_test
```
Comment (chrschng) [inline .github/workflows/build_and_test.yml:36]: I think if you use the `-//foo/bar` set operation, you need to declare the end-of-args with `--` before that.

**Review by chrschng — DISMISSED** [review]

File: .github/workflows/build_and_test.yml (hunk 4 above)
Comment (ljusten) [reply .github/workflows/build_and_test.yml]: God dammit, thanks for catching!

File: common/path_test.cc [hunk 5]
```diff
@@ -29,6 +29,12 @@
 #include "common/util.h"
 #include "gtest/gtest.h"
 
+#if PLATFORM_WINDOWS
+#define WIN32_LEAN_AND_MEAN
```
Comment (chrschng) [inline common/path_test.cc:33]: Can you wrap that in `#ifundef WIN32_LEAN_AND_MEAN` / `#endif`? Not sure if it applies here, but clangd complains if macros are redefined.

**Review by chrschng — APPROVED** [review]

File: common/path_test.cc (hunk 5 above)
Comment (ljusten) [reply common/path_test.cc:33]: Done. That should be pretty safe as we don't have any headers that define this.

**Review by chrschng — DISMISSED** [review]
Thanks!

**Review by chrschng — APPROVED** [review]

**Pull request merged**